CONTENT_ROUTING_HINTS.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+Content routing maps a CID to one or more *providers*, which specify locations where the CID can be fetched. 
+
+In order to fetch content-addressed data, there must be *some* location addressing involved. With IPFS, the implicit default starting point is a set of bootstrap nodes. (And perhaps some LAN nodes discovered by mDNS, which has a starting point of the local subnet.)
+
+So far, Kubo has planned to keep this “location addressing” implicit by adding new content routers to the default Kubo config (e.g. Filecoin indexers). But this only solves the problem for whatever specific records are provided by that indexer, and those set of implicit content routers have to be supported by the various implementations to maintain the facade of “pure content addressing”. There are also trust issues in terms of automatically sending user data to indexers that users have not explicitly trusted.
+
+Instead of gateways and IPFS nodes implicitly sending all requests to a set of content routers that changes over time, and the community needing to reach consensus on what default routers to use, this proposes specifying that the default implicit content router is *only* the IFPS public DHT and LAN DHT, and all additional content routers must be opted-in by users when making API requests.
+
+# Specification
+
+The default implicit content router for IPFS nodes is the IPFS public DHT and LAN DHT. Any additional content routers must be opted-in by users when making API requests.
+
+Users may opt-in to additional content routers using “content routing hints”, which give *suggestions* to the IPFS node about where provider records for the given CID may be found. This can include, but is not limited to, Reframe URLs, pubsub topics, multiaddrs, etc. As hints, the IPFS node is free to decide the order and strategy for using hints. If an IPFS node implements support for a hint that is specified below, it must follow the specification for that hint type.
+
+When a node receives a request with content routing hints, it should search for provider records in the IPFS public DHT and at locations specified in the hints.
+
+## Hint Types
+
+Implementations are free to support hint types that make sense for their use cases.
+
+### URI
+
+- **Reframe**
+    - HTTPS URL that ends with `/reframe` MUST be interpreted as a Reframe hint, for example:
```

Comment:
How do we specify different codecs like `dag-cbor`? Do we need to add this to the Reframe spec for HTTP URLs? And what about non-HTTP transports? Would that need to be something we add to multiaddr?

Reply:
> How do we specify different codecs like dag-cbor?

That is up to HTTP client (who controls `Accept` header sent with the request).

I think details of `/reframe` over HTTP are specified in [reframe/REFRAME_HTTP_TRANSPORT.md](https://github.com/ipfs/specs/blob/main/reframe/REFRAME_HTTP_TRANSPORT.md).
It already requires endpoint to be named `/reframe`, but we could add a paragraph to that spec which states is more explicitly. 

> what about non-HTTP transports? Would that need to be something we add to multiaddr?


We probably want to register  `/reframe` protocol at https://github.com/multiformats/multiaddr/blob/master/protocols.csv, allowing for multiaddrs like:

```
/dns4/example.net/tcp/443/tls/http/reframe
/dns4/example.net/tcp/443/wss/reframe
```

(and looping in libp2p team for sanity check, if this is correct way of representing this)